tests/test_index.py

```diff
@@ -0,0 +1,281 @@
+# Copyright 2020 CDPedistas (see AUTHORS.txt)
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 3, as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranties of
+# MERCHANTABILITY, SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR
+# PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# For further info, check  https://github.com/PyAr/CDPedia/
+
+
+import shutil
+import types
+import tempfile
+import logging
+from pprint import pprint as pp
+import pytest
+
+from src.armado import easy_index
+from src.armado import compressed_index
+
+def decomp(data):
+    """To write docset in a compact way.
+
+    Ej: 'my title/2;second title/4'"""
+    docs = [n.strip().split("/") for n in data.split(";")]
+    docs = [(n[0], int(n[1])) for n in docs]
+    return docs
+
+def abrev(result):
+    """Cut auto generated html title in 0 position."""
+    if isinstance(result, types.GeneratorType):
+        result = [r for r in result]
+    if not result:
+        return result
+    if isinstance(result, list):
```

Comment:
Es feo que esto pueda recibir "listas de elementos" o "listas de listas de elementos". Hay que simplificarlo.

Yo te diría que reciba sólo listas de elemetnos, y al usarlo abajo hagas `assert [abbrev(x) for x in res] == ...` o similar